File: two_sum/two_sum.py

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        output = []
        for i in range(len(nums)):
            num = nums[i]
            dict[num] = i

            #check if other component is in dict
            comp = target - num
            if comp in dict.keys():
                output = [dict[comp], i]
                print("1")
                return output
    
        return output


if __name__ == "__main__":
    sol = Solution()
    list = [2, 3]
    target = 5
    nums =  sol.twoSum(list, target)
    print(nums)

File: two_sum/two_sum.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

class Solution {
	public:
		std::vector<int> twoSum(std::vector<int>& nums, int target) {
			std::unordered_map<int, int> indexMap;
			std::vector<int> output(2);
			for(int i = 0; i < nums.size(); ++i) {
				int comp = target - nums[i];
				// if found
				if(indexMap.count(comp) != 0) {
					output[0] = indexMap[comp];
					output[1] = i;
					return output;
				}
				//if not found
				else {
					indexMap[nums[i]] = i;
				}
			}
			return output;
		}
};

int main() {
	std::vector<int> nums = {1, 2, 3};
	int target = 5;
	
	Solution solution;
	std::vector<int> result = solution.twoSum(nums, target);

	std::cout << "results: " << result[0] << result[1];
}
